src/jit/flowgraph.cpp

```diff
@@ -14163,16 +14163,16 @@ bool Compiler::fgOptimizeBranch(BasicBlock* bJump)
     //
     gtReverseCond(condTree);
 
+    // We need to update the following flags of the bJump block if they were set in the bbJumpDest block
+    bJump->bbFlags |= (bDest->bbFlags &
+                       (BBF_HAS_NEWOBJ | BBF_HAS_NEWARRAY | BBF_HAS_NULLCHECK | BBF_HAS_IDX_LEN | BBF_HAS_VTABREF));
+
     bJump->bbJumpKind = BBJ_COND;
     bJump->bbJumpDest = bDest->bbNext;
 
     /* Mark the jump dest block as being a jump target */
     bJump->bbJumpDest->bbFlags |= BBF_JMP_TARGET | BBF_HAS_LABEL;
```

Comment:
Similarly here, I might call this block `bDestNext` (which is defined on [13946](https://github.com/dotnet/coreclr/blob/e568823f22ec1dccc4ea785630c837e776d4ee09/src/jit/flowgraph.cpp#L13946))  in both the code (2x) and comment, rather than `bDest->bbNext`/`bJump->bbJumpDest`/`jump dest block`.